### PR TITLE
allow alternative udev

### PR DESCRIPTION
**Pull request by dkwo**
this allows to use alternative device manager as in https://github.com/void-linux/void-packages/pull/42488

### CODE REVIEW

**Comment by Duncaen** [comment]
Not a fan of this, udev is the default and I don't think we would support any other setup. You can uninstall udevd and drop in your own file if required.

**Comment by Vaelatern** [comment]
Would agree, one point of having these files in a directory instead of being a single file is so you can make these changes on your own system (or with automation) in a safe way.

**Comment by dkwo** [comment]
What you say makes sense. However, one advantage of the small modification I propose is that it kills the device manager at the end of stage 1 (if you do not start the corresponding service with runit, that it is left running currently). I also tested it and it seems to work fine.

**Comment by xplshn** [comment]
I don't see what the problem would be, none of the changes break udev, and they allow for more flexibility. And this benefits both udev and libudev-zero users

File: core-services/02-udev.sh [hunk 1]
```diff
@@ -2,18 +2,17 @@
 
 [ -n "$VIRTUALIZATION" ] && return 0
 
-if [ -x /usr/lib/systemd/systemd-udevd ]; then
-    _udevd=/usr/lib/systemd/systemd-udevd
-elif [ -x /sbin/udevd -o -x /bin/udevd ]; then
-    _udevd=udevd
-else
-    msg_warn "cannot find udevd!"
-fi
```
Comment (vincele) [inline core-services/02-udev.sh:11]: I assume loosing this systemd-specific file name and path is OK (not that I really personally care)

File: rc.conf [hunk 2]
```diff
@@ -45,3 +45,8 @@
 # RNG, for example if you plan to replicate this file system image and do not
 # have the wherewithal to first delete the contents of /var/lib/seedrng.
 #SEEDRNG_SKIP_CREDIT=false
+
+# The device manager to run during the init process. This is killed once boot
+# is complete for a supervised version to take its place. Valid values are
+# currently 'mdevd' or 'udevd' (default).
+CONFIG_DEV=udevd
```
Comment (vincele) [inline rc.conf:52]: I'd change the variable name to `CONFIG_DEVD`, or something like that, as I think the `_DEV` suffix would imply a *device*. But that's bikeshedding, feel free to ignore.

**Review by vincele — COMMENTED** [review]
LGTM

**Comment by Duncaen** [comment]
We have better things to do than having to maintain compatibility for udev alternatives for 3 users.

**Pull request closed without merging**